File: 2015/go/src/day1B.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func findFloor(floor int, valArray []int) int {
	total := 0
	foundFloor := false
	i := 0

	for i < len(valArray) && !foundFloor {
		total = total + valArray[i]
		if total == -1 {
			foundFloor = true
		}
		i++
	}

	return i
}

func convertValArrayToInt(valArray []string) []int {
	var intArray []int

	for _, val := range valArray {
		intVal, err := strconv.Atoi(val)
		check(err)
		intArray = append(intArray, intVal)
	}

	return intArray
}

func main() {
	input := readFile("day1.dat")

	//Replace Strings with numeric values
	input = strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(input, "(", "1 "), ")", "-1 "))
	valArray := strings.Split(input, " ")

	position := findFloor(-1, convertValArrayToInt(valArray))

	fmt.Println(position)
}

File: 2020/python/day1B.py
import common

def main():
    dat = common.readFile('day1.dat')
    
    val = find_pair(dat)
    print(val[0]*val[1]*val[2])


def find_pair(input):
    found = False
    i = 0
    val = (0, 0, 0)

    while (not found and i < len(input)):
        j = i + 1
        while (not found and j < len(input)):
            k = j + 1
            while (not found and k < len(input)):
                found = (int(input[i]) + int(input[j]) + int(input[k])) == 2020
                if found:
                    val = (int(input[i]), int(input[j]), int(input[k]))
                
                k = k + 1
                
            j = j + 1
        
        i = i + 1
    
    return val


if __name__ == '__main__':
    main()